cmd/containerd-shim-runhcs-v1/task_hcs.go

```diff
@@ -1008,3 +1026,72 @@ func (ht *hcsTask) requestUpdateContainer(ctx context.Context, resourcePath stri
 	}
 	return ht.c.Modify(ctx, modification)
 }
+
+// publishNotifications monitors the containers notification queue.
+// Currently only OOM events are published.
+//
+// This MUST be called via a goroutine to wait on a background thread.
+// Returned error is used for scoping, and not meant to be consummed
+func (ht *hcsTask) publishNotifications(ctx context.Context) (err error) {
+	ctx, span := trace.StartSpan(ctx, "hcsTask::publishNotifications")
+	defer span.End()
+	defer func() { oc.SetSpanStatus(span, err) }()
+	span.AddAttributes(trace.StringAttribute(logfields.TaskID, ht.id))
+	entity := log.G(ctx).WithField(logfields.TaskID, ht.id)
+
+	nq, err := ht.c.Notifications()
+	if err == notifications.ErrNotificationsNotSupported {
+		entity.Debug("container does not support notifications")
+		return
+	} else if err != nil {
+		entity.WithError(err).Warning("could not access notification queue")
+		return
+	}
```

Comment:
Now that I'm looking at it again, I don't think the distinction between what error we get matters here, as `ErrNotificationsNotSupported`'s error string already says what we need to know in it. You could just: 
```go
if err != nil {
    entity.WithError(err).Debug("unable to retrieve container notifications channel")
}
```

Reply:
Wait a sec, if this must be called in a goroutine why do we need to have this return an error?